src/lib/crypto/kimchi_backend/common/field.ml

```diff
@@ -177,24 +179,34 @@ module Make (F : Input_intf) :
             let of_sexpable = of_bigint
           end)
 
+      let zero = of_int 0
+
       let to_bignum_bigint n =
-        let rec go i two_to_the_i acc =
-          if Int.equal i size_in_bits then acc
-          else
-            let acc' =
-              if Bigint.test_bit n i then Bignum_bigint.(acc + two_to_the_i)
-              else acc
-            in
-            go (i + 1) Bignum_bigint.(two_to_the_i + two_to_the_i) acc'
-        in
-        go 0 Bignum_bigint.one Bignum_bigint.zero
-
-      let hash_fold_t s x =
-        Bignum_bigint.hash_fold_t s (to_bignum_bigint (to_bigint x))
+        (* For non-zero values, conversion is done by creating the bin_prot representation
+           of the [Bignum_bigit.t] value, and then parsing it with bin_prot. *)
+        match compare n zero with
```

Comment:
Is this necessary? The bigint representation will always be positive, so it should be fine (and actually more efficient) to just special-case zero using an equality check.

Reply:
@mrmr1993 not at all, the only purpose of all this logic is to:
1. Recover the sign because bin_prot requires it
2. Avoid bypassing bignum when converting, but if that is not an issue I can use Zarith directly

Reply:
@mrmr1993 btw, are there any values that are common and are worth short-circuiting like zero? do you think it is worth adding `one` too ?